graphbolt/src/csc_sampling_graph.cc

```diff
@@ -424,10 +445,12 @@ inline torch::Tensor NonUniformPick(
   return picked_neighbors;
 }
 
-torch::Tensor Pick(
+template <>
+torch::Tensor Pick<SamplerType::NEIGHBOR>(
     int64_t offset, int64_t num_neighbors, int64_t fanout, bool replace,
     const torch::TensorOptions& options,
-    const torch::optional<torch::Tensor>& probs_or_mask) {
+    const torch::optional<torch::Tensor>& probs_or_mask,
+    SamplerArgs<SamplerType::NEIGHBOR> args) {
```

Comment:
` SamplerArgs<SamplerType::NEIGHBOR>&`

Reply:
int64_t offset, int64_t num_neighbors are not passed by reference. This is because they are cheap to copy. You can treat this class same as those arguments. Any expensive members can be references in the class instead.

Reply:
Making it a reference will make labor version of the code slower.

Reply:
I see, it's okay to pass the argument by value . But a little confused why it is slower.

Reply:
If you look at the specializations for neighbor and labor, for neighbor the class has no members (for now) and for labor, it has 2 int64_t arguments and one reference to a torch::Tensor. The function arguments are passed by copy for int64_t values and by reference for torch::optional or potentially expensive items. If we pass the class by copy, then it will pass the two int64_t members by copy, matching the existing code, and it will pass the reference by copy, which is basically passing by reference. (passing reference by copy is like copying the pointer.). So passing the args by copy does the exact same thing as if members of it were being passed separately.

Reply:
If we pass args by reference, then it will be as if we are passing the torch::tensor by reference twice and passing int64_t values by reference, which is slower than passing by copy.